templates/vm_creation.tftpl

```diff
@@ -0,0 +1,34 @@
+%{ if create_vm_win }
+module "win_vm" {
+  source                           = "qbeyond/windows-vm/azurerm"
+  version                          = "4.1.0"
+  for_each                         = local.vm_win
+  resource_group_name              = each.value.resource_group_name
+  subnet                           = each.value.subnet
+  virtual_machine_config           = each.value.virtual_machine_config
+  admin_password                   = each.value.admin_password
+  severity_group                   = each.value.severity_group
+  update_allowed                   = each.value.update_allowed
+  data_disks                       = each.value.data_disks
+  nic_config                       = each.value.nic_config
+  public_ip_config                 = each.value.public_ip_config
+  name_overrides                   = each.value.name_overrides
+  tags                             = merge(each.value.tags, local.tags) 
+}%{ endif }
+%{ if create_vm_ux }
+module "linux_vm" {
+  source                           = "qbeyond/linux-vm/azurerm"
+  version                          = "1.0.0"
+  for_each                         = local.vm_ux
+  admin_username                   = each.value.admin_username
+  admin_credential                 = each.value.admin_credential
+  resource_group_name              = each.value.resource_group_name
+  subnet                           = each.value.subnet
+  virtual_machine_config           = each.value.virtual_machine_config
+  additional_network_interface_ids = each.value.additional_network_interface_ids
+  data_disks                       = each.value.data_disks
+  nic_config                       = each.value.nic_config
+  public_ip_config                 = each.value.public_ip_config
+  name_overrides                   = each.value.name_overrides
+  tags                             = merge(each.value.tags, local.tags) 
+}%{ endif }
```

Comment:
This is what I meant above by merging the 2 files. This just takes a local and creates a VM for each. Why not cut out the middle and just call the module directly. Then we can also add new VM modules with different versions without touching all other VMs

Reply:
Yes, I have joined the two files in one, with module and locals in the same.

We don't call the module directly because we parametrice everything, there are project that have more than one VM per subscription, and we need to managed as more as we can in a standard code. We need all the parameters included in the template, because is faster for as to use the variables that the modules have. We don't need to go everytime to the module documentation and is faster for us.